### PR TITLE
Wishing Well part 1

### PR DESCRIPTION
**Pull request by Coxswain-Navigator**
## About The Pull Request
Adds wishing well, a new original tool abnormality that lets agents and clerks gamble away their money and EGO for some sweet rewards or terrible monsters.
As of now, the abnormality will not actually be mapped in until some of the code is cleaned up and it gets a balance pass. It will, however be available for admins to spawn in if they choose to.
## Why It's Good For The Game
Adds a use for discarded EGO, another thing for clerks to do, another (technically) new abnormality.
## Changelog
:cl:
add: Wishing well tool abnormality
:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+	desc = "A well constructed of stone and wood. From where does it draw water?"
+	icon_state = "wishwell"
+	var/output
+	var/gacha
+	var/gift
+//loot lists
+	var/list/superEGO = list( //do NOT put this in the loot lists ever. SuperEGO is for inputs only so people can throw away twilight for no good reason.
+	/obj/item/ego_weapon/paradise,
+	/obj/item/clothing/suit/armor/ego_gear/twilight,
+	/obj/item/ego_weapon/twilight
```
Comment (Cupax3) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:12]: Bad indentation
```suggestion
		/obj/item/ego_weapon/paradise,
		/obj/item/clothing/suit/armor/ego_gear/twilight,
		/obj/item/ego_weapon/twilight
```

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 2]
```diff
@@ -0,0 +1,250 @@
+/obj/structure/toolabnormality/wishwell
+	name = "wishing well"
+	desc = "A well constructed of stone and wood. From where does it draw water?"
+	icon_state = "wishwell"
+	var/output
+	var/gacha
+	var/gift
+//loot lists
```
Comment (Cupax3) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:8]:
```suggestion

//loot lists
```

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+	var/list/superEGO = list( //do NOT put this in the loot lists ever. SuperEGO is for inputs only so people can throw away twilight for no good reason.
```
Comment (Cupax3) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:9]:
```suggestion

	var/list/superEGO = list( //do NOT put this in the loot lists ever. SuperEGO is for inputs only so people can throw away twilight for no good reason.
```

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+	/obj/item/ego_weapon/paradise,
+	/obj/item/clothing/suit/armor/ego_gear/twilight,
+	/obj/item/ego_weapon/twilight
+		)
+	var/list/alephitem = list(//less junk items at higher risk levels
+	/obj/item/clothing/suit/armor/ego_gear/praetorian,
+	/obj/item/toy/plush/mosb,
+	/obj/item/toy/plush/melt
```
Comment (Cupax3) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:17]:
```suggestion
		/obj/item/clothing/suit/armor/ego_gear/praetorian,
		/obj/item/toy/plush/mosb,
		/obj/item/toy/plush/melt
```

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+		)
+	var/list/wawitem = list(
+	/obj/item/toy/plush/qoh,
+	/obj/item/toy/plush/bigbird,
+	/obj/item/toy/plush/rabbit,
+	/obj/item/grenade/spawnergrenade/shrimp/super,
+	/obj/item/ego_weapon/flower_waltz
```
Comment (Cupax3) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:24]:
```suggestion
		/obj/item/toy/plush/qoh,
		/obj/item/toy/plush/bigbird,
		/obj/item/toy/plush/rabbit,
		/obj/item/grenade/spawnergrenade/shrimp/super,
		/obj/item/ego_weapon/flower_waltz
```

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+	/obj/item/paper/crumpled,
+	/obj/effect/decal/cleanable/ash,
+	/obj/item/cigbutt,
+	/obj/item/food/urinalcake,
+	/obj/item/shard,
+	/obj/item/reagent_containers/pill/maintenance
+		)
+//Potential threats
+	var/list/dawn = list(
+		/mob/living/simple_animal/hostile/ordeal/green_bot,
```
Comment (Cupax3) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:101]: like here :)

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+		/mob/living/simple_animal/hostile/slime
+		)
+
+	var/list/dusk = list(
+		/mob/living/simple_animal/hostile/ordeal/amber_dusk,
+		/mob/living/simple_animal/hostile/ordeal/green_dusk,
+		/mob/living/simple_animal/hostile/mini_censored,
+		/mob/living/simple_animal/hostile/slime/big
+		)
+//TODO: Wait until reasonable threats exist. These guys are too easy and midnights are too hard. Egor said no fixers.
```
Comment (Cupax3) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:118]:
```suggestion

//TODO: Wait until reasonable threats exist. These guys are too easy and midnights are too hard. Egor said no fixers.
```

File: code/modules/paperwork/records/info/tools.dm [hunk 8]
```diff
@@ -34,7 +34,7 @@
 	Name : Skin Prophecy<br>
 	Risk Class : Teth	<br>
 	- The employee who reads this book will experience an increase in Prudence.	<br>
-	- The book can be read until one�s Prudence reaches its limit.	<br>
+	- The book can be read until one’s Prudence reaches its limit.	<br>
```
Comment (Cupax3) [inline code/modules/paperwork/records/info/tools.dm:37]: Is this unicode character intended?

**Review by Cupax3 — CHANGES_REQUESTED** [review]
While I'm not a fan of hard-coding things, this is enough of a special system that it doesn't impact much, though an argument could be made to offload it into a define file. What is definitely not good, however, is the mixed and inconsistent indentation as well as spaces between var sections.

File: code/modules/paperwork/records/info/tools.dm (hunk 8 above)
Comment (Cupax3) [reply code/modules/paperwork/records/info/tools.dm:37]: Turns out GitHub renders `’` weird. Using standard `'` for compatibilities sake may be better (after all, BYOND runs on like Internet Explorer Ancient Edition:tm:).

Comment (Coxswain-Navigator) [reply code/modules/paperwork/records/info/tools.dm:37]: This change is not intended, I could not get a clear answer on the proper solution prior. I'll go ahead and use the standard ' as you suggested.

**Comment by Coxswain-Navigator** [comment]
> While I'm not a fan of hard-coding things, this is enough of a special system that it doesn't impact much, though an argument could be made to offload it into a define file. What is definitely not good, however, is the mixed and inconsistent indentation as well as spaces between var sections.

I originally planned on doing this, but I was advised against it. I do appreciate you taking the time to review, this should be ready for another look soon.

**Comment by Kitsunemitsu** [comment]
> While I'm not a fan of hard-coding things, this is enough of a special system that it doesn't impact much, though an argument could be made to offload it into a define file. What is definitely not good, however, is the mixed and inconsistent indentation as well as spaces between var sections.

Hardcoding it would require all coders to update it every time they added in a weapon or armor set.

**Review by Coxswain-Navigator — COMMENTED** [review]
All issues should be addressed as of now.

**Comment by Cupax3** [comment]
> > While I'm not a fan of hard-coding things, this is enough of a special system that it doesn't impact much, though an argument could be made to offload it into a define file. What is definitely not good, however, is the mixed and inconsistent indentation as well as spaces between var sections.
> 
> Hardcoding it would require all coders to update it every time they added in a weapon or armor set.

Depends on the scope of the abnormality, which I'm sadly completely unknown on.
From your comment I take it that the abnormality is supposed to be able to draw from any EGO weapon or armor?

**Comment by Coxswain-Navigator** [comment]
> Depends on the scope of the abnormality, which I'm sadly completely unknown on. From your comment I take it that the abnormality is supposed to be able to draw from any EGO weapon or armor?

There is a layer of complexity here. You will notice that the lists are created with hardcoded values in them, yet every EGO with a cost associated with it is added to the lists through code. The hardcoded parts are intentional - they are to be manually adjusted as I recieve feedback, whereas the EGO part is automated and I don't have to manually add every new EGO.

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+		/obj/item/clothing/suit/hawaiian
+		)
+	var/list/tethitem = list(
+		/obj/item/clothing/suit/armor/ego_gear/training,
+		/obj/item/ego_weapon/training,
+		/obj/item/clothing/suit/armor/ego_gear/rookie,
+		/obj/item/clothing/suit/armor/ego_gear/fledgling,
+		/obj/item/clothing/suit/armor/ego_gear/apprentice,
+		/obj/item/clothing/suit/armor/ego_gear/freshman,
+		/obj/item/clothing/under/suit/lobotomy/extraction/arbiter,
```
Comment (Kitsunemitsu) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:44]: Make this the regular extraction suit.

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+		/obj/item/food/bread/bongbread,
+		/obj/item/clothing/mask/cigarette/cigar/havana
+		)
+	var/list/normalitem = list(
+		/obj/item/reagent_containers/hypospray/medipen/salacid,
+		/obj/item/reagent_containers/hypospray/medipen/mental,
+		/obj/item/ego_weapon/tutorial,
+		/obj/item/ego_weapon/tutorial/white,
+		/obj/item/ego_weapon/tutorial/black,
+		/obj/item/ego_weapon/tutorial/pale,
```
Comment (Kitsunemitsu) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:61]: I would say "It's not smart to give players pale but honestly this is 7 pale IIRC it should be ok

**Review by Kitsunemitsu — DISMISSED** [review]
Code looks good actually.
Approved

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm (hunk 9 above)
Comment (Coxswain-Navigator) [reply code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:44]: Intended as far as I am concerned, take a look at the normal list to see what I mean. If you feel that it is in a poor place, I can always move it to a higher risk level. At the time, I don't want too many non-EGO in Waw or Aleph though.

**Review by Cupax3 — APPROVED** [review]
I can't find anything to complain about so if this is tested, it's fine

**Comment by Cupax3** [comment]
Please squash before merge tho

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm (hunk 9 above)
Comment (EgorDinamit) [reply code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:44]: We're asking you to not put a literal arbiter outfit there.

**Review by Kitsunemitsu — APPROVED** [review]
I meant to give my approval before, I am again now.

**Comment by Kitsunemitsu** [comment]
Remove attacking it with weapons and I will merge

File: code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+			if (35 to 40)
+				heitem+=ego.item_path
+			if (20 to 25)
+				tethitem+=ego.item_path
+			if (12)
+				zayinitem+=ego.item_path
+
+//End of loot lists
+/obj/structure/toolabnormality/wishwell/attackby(obj/item/I, mob/living/carbon/human/user)
+	..()
```
Comment (Kitsunemitsu) [inline code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:148]: Remove this line

Comment (Coxswain-Navigator) [reply code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:148]: Removing this line causes far more issues than it solves, but I can try to find an alternate solution.

Comment (Coxswain-Navigator) [reply code/modules/mob/living/simple_animal/abnormality/_tools/zayin/wishwell.dm:148]: After testing the change doesn't seem feasible with respect to the rest of the code at the time of writing.

**Pull request merged**